File: day1/main.nim
import strutils       # splitLines 
import std/sugar      # for collect
import std/algorithm  # for sort
import std/math       # for sum

proc readLines(): seq[string] =
    return readFile("input").splitLines()

iterator calcCalories(): int =
    let lines = readLines()
    var s = 0
    for line in lines:
        if line != "":
            s += line.parseInt()
        else:
            yield s
            s = 0
    if s != 0:
        yield s

proc part1() =
    let cals = collect(newSeq):
        for c in calcCalories(): c
    echo max(cals)

proc part2() =
    var cals = collect(newSeq):
        for c in calcCalories(): c
    cals.sort(SortOrder.Descending)
    echo sum(cals[0 ..< 3])

part1()
part2()

File: day1/main.py
from utils.utils import readInputLines

def caloriesPerElf(lines):
    cals = []
    for line in lines:
        if line != "":
            cals.append(int(line))
        else:
            yield sum(cals)
            cals = []
    if len(cals) != 0:
        yield sum(cals)

def part1():
    lines = readInputLines(__file__)
    sums = caloriesPerElf(lines)
    print(max(sums))

def part2():
    lines = readInputLines(__file__)
    sums = caloriesPerElf(lines)
    sortedSums = sorted(sums, reverse=True)
    top3 = sortedSums[0:3]
    print(sum(top3))

if __name__ == "__main__":
    part1()
    part2()